### PR TITLE
Java 9+ compatibility

### LINKED ISSUE #259: --reporter/--ruleset Java 9+ compatibility
I checked out the repo and ran `./gradlew build` in `ktlint-ruleset-template` to generate the jars. I then tried running ktlint with all of those jars against a kotlin class and received

``` java
Exception in thread "main" java.lang.ClassCastException: java.base/jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to java.base/java.net.URLClassLoader
	at com.github.shyiko.ktlint.Main.loadJARs(Main.kt:567)
	at com.github.shyiko.ktlint.Main.main(Main.kt:253)
```

I can't see anything wrong with the jar structure or the paths I am using, but I'm guessing I have done something wrong.

The command I used was:
`ktlint -R build/libs/ktlint-ruleset-template.jar src/main/kotlin/yourpkgname/NoVarRule.kt`

I also tried using the sources jar.

### PR DESCRIPTION
**Pull request by marschwar**
This PR attempts to resolve #259.

It looks like much more than it actually is. Essentially I changed.

```
(ClassLoader.getSystemClassLoader() as java.net.URLClassLoader)
    .addURLs(artifactUrls)
```
to
```
URLClassLoader(artifactUrls.toTypedArray(), parentClassloader)
```

All the other changes are basically refactoring to be able to write a test for it. I tried to make small commits so there is a chance for someone else to review the changes. Unfortunately I find it really hard to make sure I did not break anything.

One more thing: In order to test that a custom ruleset is downloaded via maven, I added the first one I could find (https://github.com/gabrielittner/ktlint-rules). This obviously is not ideal. I hope someone else has a better idea.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ skip_tags: true
 environment:
```
Comment (jaredsburrows) [inline .appveyor.yml]: This file is no longer in master. Can you remove this file?

**Comment by jaredsburrows** [comment]
If you are attempting to solve #259, can you remove changes that are unrelated such as upgrading gradle?

**Comment by marschwar** [comment]
Sure. As there was no feedback up until now I did not bother to rebase onto master. If you think it is going in the right direction, I would be happy to do so. If not I would close the PR.

**Comment by jaredsburrows** [comment]
Usually, larger the PRs or greater the impact may take more time. Keeping the change small and specific will help code reviews go faster :).

**Comment by jacquerie** [comment]
According to https://blog.codefx.org/java/java-9-migration-guide/#Casting-To-URL-Class-Loader, this is not completely straightforward:
> If you’ve used the URLClassLoader to dynamically load user provided code (for example as part of a plugin infrastructure) by appending to the class path, then you have to find a new way to do that as it can not be done with Java 9.

So, perhaps, some additional complexity introduced by this PR might be warranted. I can take care of rebasing and removing extraneous stuff such as style changes and version bumps!

**Pull request closed without merging**